packages/shared/src/components/filters/FeedFilters.tsx

```diff
@@ -20,60 +26,78 @@ interface FeedFiltersProps {
   onBack?: () => void;
 }
 
+interface UnblockItem {
+  tag?: string;
+  source?: Source;
+  action?: () => unknown;
+}
+
 export default function FeedFilters({
   isOpen,
   onBack,
 }: FeedFiltersProps): ReactElement {
+  const client = useQueryClient();
+  const [unblockItem, setUnblockItem] = useState<UnblockItem>();
+  const { user } = useContext(AuthContext);
   const { alerts, updateAlerts } = useContext(AlertContext);
   const { hasAnyFilter } = useFeedSettings();
-  const [hidden, setHidden] = useState(true);
-  const timeoutRef = useRef<number>();
+  const { flags } = useContext(FeaturesContext);
+  const shouldShowMyFeed = isFeaturedEnabled(Features.MyFeedOn, flags);
 
   useEffect(() => {
     if (isOpen) {
-      if (alerts?.filter && hasAnyFilter) {
+      if (alerts?.filter && hasAnyFilter && user) {
         updateAlerts({ filter: false });
       }
-
-      if (!hidden) {
-        return;
-      }
-
-      if (timeoutRef.current) clearTimeout(timeoutRef.current);
-      setHidden(false);
-      return;
     }
+  }, [isOpen, alerts, user, hasAnyFilter]);
 
-    if (hidden) {
-      return;
-    }
-
-    timeoutRef.current = window.setTimeout(() => setHidden(true), 300);
-    // eslint-disable-next-line consistent-return
-    return () => timeoutRef.current && clearTimeout(timeoutRef.current);
-  }, [isOpen, alerts, hasAnyFilter]);
+  const onCreate = () => {
+    // apply analytics
+    const key = getFeedSettingsQueryKey(user);
+    const { feedSettings } = client.getQueryData(key) as AllTagCategoriesData;
+    updateLocalFeedSettings(feedSettings);
+    window.location.replace(`${process.env.NEXT_PUBLIC_WEBAPP_URL}register`);
```

Comment:
We need to open the login modal. We can't redirect to registration directly.

Reply:
Oh, my mistake. Let me rectify this.